### PR TITLE
#109: Create additional validation step for other objects

### PR DESCRIPTION
**Pull request by yefei0423**
Signed-off-by: Fei Ye <feiye@cn.ibm.com>

### CODE REVIEW

**Review by fketelaars — COMMENTED** [review]
Hit the following error:
```
TASK [lint-config : Run the linter and pre-processor script for object nfs_server] ***
Monday 12 September 2022  07:54:12 +0000 (0:00:00.041)       0:00:18.433 ******
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "python3 pre-execution-processor.py  -k nfs_server  -p /cloud-pak-deployer/automation-generators/aws/nfs_server  -i 0  -s preprocessor.py  -a aW5mcmFzdHJ1Y3R1cmU6IHthd3NfcmVnaW9uOiBldS13ZXN0LTJ9Cm5hbWU6IGZrLXJvc2EKb3BlbnNoaWZ0X2NsdXN0ZXJfbmFtZTogZmstcm9zYQo=  -f Z2xvYmFsX2NvbmZpZzoge2F3c19yZWdpb246IGV1LXdlc3QtMSwgY2xvdWRfcGxhdGZvcm06IGF3cywgZW52aXJvbm1lbnRfbmFtZTogc2FtcGxlfQpuZnNfc2VydmVyOgotIGluZnJhc3RydWN0dXJlOiB7YXdzX3JlZ2lvbjogZXUtd2VzdC0yfQogIG5hbWU6IGZrLXJvc2EKICBvcGVuc2hpZnRfY2x1c3Rlcl9uYW1lOiBmay1yb3NhCm9wZW5zaGlmdDoKLSBjbG91ZF9uYXRpdmVfdG9vbGtpdDogdHJ1ZQogIGNvbXB1dGVfZmxhdm91cjogbTUueGxhcmdlCiAgY29tcHV0ZV9ub2RlczogMwogIGluZnJhc3RydWN0dXJlOiB7YXdzX3JlZ2lvbjogZXUtd2VzdC0yLCBtdWx0aV96b25lOiB0cnVlLCB0eXBlOiByb3NhLCB1c2Vfc3RzOiBmYWxzZX0KICBuYW1lOiBmay1yb3NhCiAgb2NwX3ZlcnNpb246IDQuOC40NAogIG9wZW5zaGlmdF9zdG9yYWdlOgogIC0ge25mc19zZXJ2ZXJfbmFtZTogZmstcm9zYSwgc3RvcmFnZV9uYW1lOiBhd3Mtc3RvcmFnZSwgc3RvcmFnZV90eXBlOiBhd3MtZWxhc3RpY30KLSBjb21wdXRlX2ZsYXZvdXI6IG01LnhsYXJnZQogIGNvbXB1dGVfbm9kZXM6IDMKICBjb250cm9sX3BsYW5lX2ZsYXZvdXI6IG00LnhsYXJnZQogIGNvbnRyb2xfcGxhbmVfbm9kZXM6IDMKICBkb21haW5fbmFtZTogZmtkZW1vcy5ubAogIGluZnJhc3RydWN0dXJlOiB7YXdzX3JlZ2lvbjogZXUtd2VzdC0xLCBtdWx0aV96b25lOiB0cnVlLCB0eXBlOiBzZWxmLW1hbmFnZWQsIHVzZV9zdHM6IGZhbHNlfQogIG5hbWU6IGZrLWF3cwogIG9jcF92ZXJzaW9uOiA0LjguNDQKICBvcGVuc2hpZnRfc3RvcmFnZToKICAtIHtvY3Nfc3RvcmFnZV9sYWJlbDogb2NzLCBvY3Nfc3RvcmFnZV9zaXplX2diOiA1MDAsIHN0b3JhZ2VfbmFtZTogb2NzLXN0b3JhZ2UsCiAgICBzdG9yYWdlX3R5cGU6IG9jc30K  -t aws  -v X2F3c19hY2Nlc3Nfa2V5OiBBS0lBMkZMTVBBV0JNSlFRRDVHUApfYXdzX3NlY3JldF9hY2Nlc3Nfa2V5OiA0a0h3K0Y3cUN6aWJNcXIwb2liVEkwenl4SVJtUEdYbTNnWXUweWxRCl9vY3BfcHVsbHNlY3JldDoKICBhdXRoczoKICAgIGNsb3VkLm9wZW5zaGlmdC5jb206IHthdXRoOiBiM0JsYm5Ob2FXWjBMWEpsYkdWaGMyVXRaR1YySzI5amJWOWhZMk5sYzNOZk0yVTNNREEyWWpJeFpXVmlORE5pT1Roak5EVTVNV05oWlRaaE1HUTBOR1E2TVRWUVRWWTJObEJHUXpoVlIwWkhRVVpPUVRoYU5UVkVOVXc1TjFkS01WVlpOVWRQTnpSTE9GZEVRMGN3V2xwQ1JrSlRSa0pPTURWWFV6RlBNMFZDUlE9PSwKICAgICAgZW1haWw6IGZrZXRlbGFhcnNAbmwuaWJtLmNvbX0KICAgIHF1YXkuaW86IHthdXRoOiBiM0JsYm5Ob2FXWjBMWEpsYkdWaGMyVXRaR1YySzI5amJWOWhZMk5sYzNOZk0yVTNNREEyWWpJeFpXVmlORE5pT1Roak5EVTVNV05oWlRaaE1HUTBOR1E2TVRWUVRWWTJObEJHUXpoVlIwWkhRVVpPUVRoYU5UVkVOVXc1TjFkS01WVlpOVWRQTnpSTE9GZEVRMGN3V2xwQ1JrSlRSa0pPTURWWFV6RlBNMFZDUlE9PSwKICAgICAgZW1haWw6IGZrZXRlbGFhcnNAbmwuaWJtLmNvbX0KICAgIHJlZ2lzdHJ5LmNvbm5lY3QucmVkaGF0LmNvbToge2F1dGg6IGZIVm9ZeTF3YjI5c0xXWXhaV0U1WVdVNExUWTNOMlV0TkRnNFppMDRZVGsxTFRVMU9EVmlZV0k1WldabE5qcGxlVXBvWWtkamFVOXBTbE5WZWxWNFRXbEtPUzVsZVVwNlpGZEphVTlwU1hkT1ZHUnNUa1JCZDA0eVNtdGFSMGt3V2xSbk5WbFhTVE5PUkZwb1dsUnNhbHBxVm14UFZFVXhUVk5LT1M1clJsZzNSWEEwU2kxUGJETktaMFZNY0daYU1WWjJSMEUwVW1kVlJDMTRNbVl6ZEhGSWFtTnhlbmRsWkZZMFZYUTJkemxsZVd4clExQlRhMU5ETTI4dE0wRm9PVXM0VEdkWll5MTRWMW8wZUV4UE5HcEhXbXBmU2xOc05HRTBVREZhVG5KVlVYVldkMnRYTmpGbE5VRXpTM1p0TjNKWVYyWllYelJDUVhaMFpYbGlUVjlaTFRCak9GSnZaM0pFVVd4amFHTnpRWGQzZVhFMmNWQlZXVU0yWkV4NGVHNXdiMFpsT0ZSS2FXMU1UMEZ5YW5GT1N6VkdSMkpGTW04ek9YVlBNRzlqV0ZrM1FXczFYMDV4WmsxR2VXOXZhM0V4YURSU2RGWlVVak5uVURSNVJXaHhVVzlXVEZsMmQxQlFZV0ZDV25nM1pUZG1hVFIxVERoT1ZYZFVjbEZUTUhKTE4zaGhNbTFvU0Zoek9GQTBRMVJMTUhOYVRWWjNYMUo1U0RoV01IVk9hSEZPWTJkNlRHOXFNbGh5WkhZdE1HVmFRMUpqUjJaU2RVdGxSMGxETkRsUVowaG9UVlZXYzNWelIzcERXa0prT1ZKemNsQllOWFJLTTFOMVZIcFVaVkpPVkVnemRWUnJkbFJ1ZEd0Uk9XMTFVbWxIT1d4SWRrRTBUbTlKVGtWeWVtdG9iamhQVEVoc1NHSnhTRmxXUVhCRk0wTlJOREZVZWpaT1lraHRkalpzZG01TVZWUTJXbWxoYVhKSmFYVllRMG8xTVZCdlEycFRNM1l4YjJaV1VYTkJUV0Y0TkdrNGRsTmxVRzFmVkV4TVpWWlFNV0ZuZUVOUVgxRkNOSEZIVUUwd09XOUlTRnBqYkdSVVNIYzVVelZGUzBaMFVHZGZlV1p3VmpaaVNHVkNUWFZ0VERJMWVHOXdXREJSWDFoMlpXUkxiVzl4TFZadlMxSktUSEJqWkZCVmNUWjBXVlZqY1dsVmFHMDVkelk0VDNGc1RXNDFPVE10V1Rkb1JXcGtaR2RwZWxJNVEzVjJSR2xoY1ROZk5tVmZWRU51VUdSd1JUaHdlWEl3U0dkcmJHVnJka2RFU2xCTGRHSjZibWxQYWpaeldFc3laWEpPTjBKVk1IWlJXSFpEYVVScGNUZFpSbk5EU1Rka1VEQXRNR2hoWW1Gbk1IbEVabmhhUjJWZmFqbEVaVnBUYnc9PSwKICAgICAgZW1haWw6IGZrZXRlbGFhcnNAbmwuaWJtLmNvbX0KICAgIHJlZ2lzdHJ5LnJlZGhhdC5pbzoge2F1dGg6IGZIVm9ZeTF3YjI5c0xXWXhaV0U1WVdVNExUWTNOMlV0TkRnNFppMDRZVGsxTFRVMU9EVmlZV0k1WldabE5qcGxlVXBvWWtkamFVOXBTbE5WZWxWNFRXbEtPUzVsZVVwNlpGZEphVTlwU1hkT1ZHUnNUa1JCZDA0eVNtdGFSMGt3V2xSbk5WbFhTVE5PUkZwb1dsUnNhbHBxVm14UFZFVXhUVk5LT1M1clJsZzNSWEEwU2kxUGJETktaMFZNY0daYU1WWjJSMEUwVW1kVlJDMTRNbVl6ZEhGSWFtTnhlbmRsWkZZMFZYUTJkemxsZVd4clExQlRhMU5ETTI4dE0wRm9PVXM0VEdkWll5MTRWMW8wZUV4UE5HcEhXbXBmU2xOc05HRTBVREZhVG5KVlVYVldkMnRYTmpGbE5VRXpTM1p0TjNKWVYyWllYelJDUVhaMFpYbGlUVjlaTFRCak9GSnZaM0pFVVd4amFHTnpRWGQzZVhFMmNWQlZXVU0yWkV4NGVHNXdiMFpsT0ZSS2FXMU1UMEZ5YW5GT1N6VkdSMkpGTW04ek9YVlBNRzlqV0ZrM1FXczFYMDV4WmsxR2VXOXZhM0V4YURSU2RGWlVVak5uVURSNVJXaHhVVzlXVEZsMmQxQlFZV0ZDV25nM1pUZG1hVFIxVERoT1ZYZFVjbEZUTUhKTE4zaGhNbTFvU0Zoek9GQTBRMVJMTUhOYVRWWjNYMUo1U0RoV01IVk9hSEZPWTJkNlRHOXFNbGh5WkhZdE1HVmFRMUpqUjJaU2RVdGxSMGxETkRsUVowaG9UVlZXYzNWelIzcERXa0prT1ZKemNsQllOWFJLTTFOMVZIcFVaVkpPVkVnemRWUnJkbFJ1ZEd0Uk9XMTFVbWxIT1d4SWRrRTBUbTlKVGtWeWVtdG9iamhQVEVoc1NHSnhTRmxXUVhCRk0wTlJOREZVZWpaT1lraHRkalpzZG01TVZWUTJXbWxoYVhKSmFYVllRMG8xTVZCdlEycFRNM1l4YjJaV1VYTkJUV0Y0TkdrNGRsTmxVRzFmVkV4TVpWWlFNV0ZuZUVOUVgxRkNOSEZIVUUwd09XOUlTRnBqYkdSVVNIYzVVelZGUzBaMFVHZGZlV1p3VmpaaVNHVkNUWFZ0VERJMWVHOXdXREJSWDFoMlpXUkxiVzl4TFZadlMxSktUSEJqWkZCVmNUWjBXVlZqY1dsVmFHMDVkelk0VDNGc1RXNDFPVE10V1Rkb1JXcGtaR2RwZWxJNVEzVjJSR2xoY1ROZk5tVmZWRU51VUdSd1JUaHdlWEl3U0dkcmJHVnJka2RFU2xCTGRHSjZibWxQYWpaeldFc3laWEpPTjBKVk1IWlJXSFpEYVVScGNUZFpSbk5EU1Rka1VEQXRNR2hoWW1Gbk1IbEVabmhhUjJWZmFqbEVaVnBUYnc9PSwKICAgICAgZW1haWw6IGZrZXRlbGFhcnNAbmwuaWJtLmNvbX0KX29jcF9zc2hfcHViX2tleTogJ3NzaC1yc2EgQUFBQUIzTnphQzF5YzJFQUFBQURBUUFCQUFBQmdRQ1o1cjc3M1RiYWs4MUVZT21OOFlPN1JYWDFKTzM3Sm55VFNCWXhzKzZ6OVRtYkRjZ1k2aUZraSs1YVlCbkphWW1rS2xRNXV0RXcwWmUrY2NYSUh4eW44MzFFT1NrZmlVdks3OHVJRkNKRHNvbG92Qk9KNklQaVIwNUY1dCtEbDhUd3BKSExaTmFvVFJIUkNOUEhNSzVzYnliaXAxcVBpb213dkhHVmd2RTAwMmtXTXdPbFpqR3pjRWJJeXJvZVhQckE4ZmY5TUJ5cGsxdFpwQVRuNCttNHhtQ1JqblIxRXlQUm9qRHdBQnd4Uy9lbEg4aEcxeHZUcGFlMzZGMEdrTGovM0RFdDZQZkFKV25yMW5WeS9adC9hVGxPcFVBbGVuaFVxbE43MlBvbkRBbWxmQzhUQWdNTDd1MWFMZloxUXNUdnFySGh3aTE3dTZ4ejN3VjJReXRpUndjVkluK1g1R1JFcTJZRUwxOG1taXd1WVFpdGd5R3lMSnRONGFWb0tFWXNkOWY3N1ZRV1dUUGJibVluaWtzY1d0bGFUMThKVXFyeko5NC9PWUJxb3ZKdHNSZFJjQ2dnL3VCTGl1WVM3ODhnVXBHeWhZRXlhM3c1QmV6MklLK2MxVkJ4OGI0ZFpPWjE1cGJCbDJnQU42S3EwR0FGVTdkRUd0QUNhRDdMWFdFPQogIGZrZXRlbGFhcnNAYnJpbmd1cC1pYm0KCiAgJwpfcm9zYV9sb2dpbl90b2tlbjogZXlKaGJHY2lPaUpJVXpJMU5pSXNJblI1Y0NJZ09pQWlTbGRVSWl3aWEybGtJaUE2SUNKaFpEVXlNamRoTXkxaVkyWmtMVFJqWmpBdFlUZGlOaTB6T1RrNE16VmhNRGcxTmpZaWZRLmV5SnBZWFFpT2pFMk5qRXpOVEk1T1RVc0ltcDBhU0k2SWpZMU56Um1aRE5oTFdFMVpEVXROR0V4WVMwNE5qQTRMV05rWkRaak1qRXhZV1ZtWXlJc0ltbHpjeUk2SW1oMGRIQnpPaTh2YzNOdkxuSmxaR2hoZEM1amIyMHZZWFYwYUM5eVpXRnNiWE12Y21Wa2FHRjBMV1Y0ZEdWeWJtRnNJaXdpWVhWa0lqb2lhSFIwY0hNNkx5OXpjMjh1Y21Wa2FHRjBMbU52YlM5aGRYUm9MM0psWVd4dGN5OXlaV1JvWVhRdFpYaDBaWEp1WVd3aUxDSnpkV0lpT2lKbU9qVXlPR1EzTm1abUxXWTNNRGd0TkRObFpDMDRZMlExTFdabE1UWm1OR1psTUdObE5qcG1hMlYwWld4aFlYSnpMV2xpYlNJc0luUjVjQ0k2SWs5bVpteHBibVVpTENKaGVuQWlPaUpqYkc5MVpDMXpaWEoyYVdObGN5SXNJbTV2Ym1ObElqb2lNR1ExTmpRMlpXRXRZemc0TVMwME9USXpMVGxqTURVdE1UQXlNVFEwTXpRNFpqTXpJaXdpYzJWemMybHZibDl6ZEdGMFpTSTZJbVV3WWpSallUSmtMVGRsWkRRdE5EazBaUzFpTldaaExUUTVaamsxTmpKaE5EbG1aQ0lzSW5OamIzQmxJam9pYjNCbGJtbGtJR0Z3YVM1cFlXMHVjMlZ5ZG1salpWOWhZMk52ZFc1MGN5QnZabVpzYVc1bFgyRmpZMlZ6Y3lJc0luTnBaQ0k2SW1Vd1lqUmpZVEprTFRkbFpEUXRORGswWlMxaU5XWmhMVFE1WmprMU5qSmhORGxtWkNKOS5GQ0JuZ1dUM0NhbkxmbWlNMUduT2JJQ1laZ3kyZkRQa2w5eTVhbk1BTFFBCl9zZWNyZXRfdmFsdWVfYjY0OiBaWGxLYUdKSFkybFBhVXBKVlhwSk1VNXBTWE5KYmxJMVkwTkpaMDlwUVdsVGJHUlZTV2wzYVdFeWJHdEphVUUyU1VOS2FGcEVWWGxOYW1Sb1RYa3hhVmt5V210TVZGSnFXbXBCZEZsVVpHbE9hVEI2VDFSck5FMTZWbWhOUkdjeFRtcFphV1pSTG1WNVNuQlpXRkZwVDJwRk1rNXFSWHBPVkVrMVQxUlZjMGx0Y0RCaFUwazJTV3BaTVU1NlVtMWFSRTVvVEZkRk1WcEVWWFJPUjBWNFdWTXdORTVxUVRSTVYwNXJXa1JhYWsxcVJYaFpWMVp0V1hsSmMwbHRiSHBqZVVrMlNXMW9NR1JJUW5wUGFUaDJZek5PZGt4dVNteGFSMmhvWkVNMWFtSXlNSFpaV0ZZd1lVTTVlVnBYUm5OaVdFMTJZMjFXYTJGSFJqQk1WMVkwWkVkV2VXSnRSbk5KYVhkcFdWaFdhMGxxYjJsaFNGSXdZMGhOTmt4NU9YcGpNamgxWTIxV2EyRkhSakJNYlU1MllsTTVhR1JZVW05TU0wcHNXVmQ0ZEdONU9YbGFWMUp2V1ZoUmRGcFlhREJhV0VwMVdWZDNhVXhEU25wa1YwbHBUMmxLYlU5cVZYbFBSMUV6VG0xYWJVeFhXVE5OUkdkMFRrUk9iRnBETURSWk1sRXhURmRhYkUxVVdtMU9SMXBzVFVkT2JFNXFjRzFoTWxZd1dsZDRhRmxZU25wTVYyeHBZbE5KYzBsdVVqVmpRMGsyU1dzNWJWcHRlSEJpYlZWcFRFTkthR1Z1UVdsUGFVcHFZa2M1TVZwRE1YcGFXRW95WVZkT2JHTjVTWE5KYlRWMlltMU9iRWxxYjJsTlIxRXhUbXBSTWxwWFJYUlplbWMwVFZNd01FOVVTWHBNVkd4cVRVUlZkRTFVUVhsTlZGRXdUWHBSTkZwcVRYcEphWGRwWXpKV2VtTXliSFppYkRsNlpFZEdNRnBUU1RaSmJWVjNXV3BTYWxsVVNtdE1WR1JzV2tSUmRFNUVhekJhVXpGcFRsZGFhRXhVVVRWYWFtc3hUbXBLYUU1RWJHMWFRMGx6U1c1T2FtSXpRbXhKYW05cFlqTkNiR0p0Ykd0SlIwWjNZVk0xY0ZsWE1IVmpNbFo1Wkcxc2FscFdPV2haTWs1MlpGYzFNR041UW5aYWJWcHpZVmMxYkZneVJtcFpNbFo2WTNsSmMwbHVUbkJhUTBrMlNXMVZkMWxxVW1wWlZFcHJURlJrYkZwRVVYUk9SR3N3V2xNeGFVNVhXbWhNVkZFMVdtcHJNVTVxU21oT1JHeHRXa05LT1M1R1EwSnVaMWRVTTBOaGJreG1iV2xOTVVkdVQySkpRMWxhWjNreVprUlFhMnc1ZVRWaGJrMUJURkZCCg==\n", "delta": "0:00:00.380662", "end": "2022-09-12 07:54:13.571187", "msg": "non-zero return code", "rc": 1, "start": "2022-09-12 07:54:13.190525", "stderr": "Traceback (most recent call last):\n  File \"pre-execution-processor.py\", line 36, in <module>\n    result = preprocessor(attributes=generatorAttributes, fullConfig=generatorFullConfig, moduleVariables=generatorVariables)\nTypeError: preprocessor() got an unexpected keyword argument 'moduleVariables'", "stderr_lines": ["Traceback (most recent call last):", "  File \"pre-execution-processor.py\", line 36, in <module>", "    result = preprocessor(attributes=generatorAttributes, fullConfig=generatorFullConfig, moduleVariables=generatorVariables)", "TypeError: preprocessor() got an unexpected keyword argument 'moduleVariables'"], "stdout": "", "stdout_lines": []}
```

File: automation-roles/10-validation/lint-config/scripts/pre-execution-processor.py [hunk 1]
```diff
@@ -28,8 +32,10 @@
 # attributes_updated: <dict>
 # errors: []
 
-
-result = preprocessor(attributes=generatorAttributes, fullConfig=generatorFullConfig)
+if args.platform == "aws":
```
Comment (fketelaars) [inline automation-roles/10-validation/lint-config/scripts/pre-execution-processor.py:35]: The extra parameters should not be dependent on the cloud platform. If required update all `preprocessor.py`  scripts to accept the additional parameter, even if not used.

**Review by fketelaars — COMMENTED** [review]
When deployer is run with `--check-only`, it fails with:
```
TASK [generators : Gather information about all availabily zones for the region eu-west-1] ***
Monday 12 September 2022  08:05:23 +0000 (0:00:00.071)       0:00:26.105 ******
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "aws ec2 describe-availability-zones  --region eu-west-1 \n", "delta": "0:00:00.006267", "end": "2022-09-12 08:05:23.397688", "msg": "non-zero return code", "rc": 127, "start": "2022-09-12 08:05:23.391421", "stderr": "/bin/sh: aws: command not found", "stderr_lines": ["/bin/sh: aws: command not found"], "stdout": "", "stdout_lines": []}
```

**Review by fketelaars — CHANGES_REQUESTED** [review]
Please make the changes indicated in the PR review.

File: automation-roles/10-validation/lint-config/scripts/pre-execution-processor.py (hunk 1 above)
Comment (yefei0423) [reply automation-roles/10-validation/lint-config/scripts/pre-execution-processor.py:35]: @fketelaars , Thanks. I will take a look at the issue you found, and update all `preprocessor.py` scripts. `preprocessor.py` script will only apply for new validation way for aws.

Comment (fketelaars) [reply automation-roles/10-validation/lint-config/scripts/pre-execution-processor.py]: Please make sure it applies to all validations, so update all preprocessor.py scripts and correct the parameters.

Comment (yefei0423) [reply automation-roles/10-validation/lint-config/scripts/pre-execution-processor.py]: @fketelaars, now only aws platform will apply for new validations. For other platform, it still apply for previous validations even if I updated all preprocessor.py scripts to accept one more parameter.   If aws new validations are expected, I will update code for other platforms.  Thanks.

Comment (fketelaars) [reply automation-roles/10-validation/lint-config/scripts/pre-execution-processor.py]: Looks good now. Thanks.

**Review by fketelaars — APPROVED** [review]

**Pull request merged**